### PR TITLE
[Merged by Bors] - Fix genesis state download panic when running in debug mode

### PR DESCRIPTION
**Pull request by jimmygchen**
## Issue Addressed

#4738 

## Proposed Changes

See the above issue for details. Went with option #2 to use the async reqwest client in `Eth2NetworkConfig` and propagate the async-ness.

### CODE REVIEW

File: boot_node/src/lib.rs [hunk 1]
```diff
@@ -81,20 +81,13 @@ fn main<T: EthSpec>(
...
+    // Run the boot node
+    runtime.block_on(server::run::<T>(
         lh_matches,
-        &config_sz,
-        &eth2_network_config.chain_spec::<T>()?,
-    )?;
+        bn_matches,
+        eth2_network_config,
+        log,
+    ))?;
```
Comment (jimmygchen) [inline boot_node/src/lib.rs:90]: Move config parsing and dumping into `server::run` as they need to be async.

File: common/eth2_network_config/src/lib.rs [hunk 2]
```diff
@@ -149,10 +144,8 @@ impl Eth2NetworkConfig {
                     )
                 })
         } else {
-            self.genesis_state::<E>(genesis_state_url, timeout, log)?
-                .map(|state| state.genesis_validators_root())
-                .map(Result::Ok)
-                .transpose()
+            self.get_genesis_state_from_bytes::<E>()
```
Comment (jimmygchen) [inline common/eth2_network_config/src/lib.rs:147]: Switching to the new sync `get_genesis_state_from_bytes` method, as we don't need to download genesis state and `genesis_state` method is now async.

**Comment by michaelsproul** [comment]
Looks like there's a small type issue in the `boot_node`: https://github.com/sigp/lighthouse/actions/runs/6232160760/job/16930311594?pr=4753#step:4:1310

**Comment by jimmygchen** [comment]
> Looks like there's a small type issue in the `boot_node`: https://github.com/sigp/lighthouse/actions/runs/6232160760/job/16930311594?pr=4753#step:4:1310

Thanks! this one is slightly annoying as it's only an issue on the `msrv` - I'll look into it.

**Comment by jimmygchen** [comment]
I've just fixed a copy paste bug I introduced ...tests should pass now 🤞

**Comment by jimmygchen** [comment]
I've tested running this branch on `--network holesky` in debug mode, and can confirm that it no longer panics.

**Comment by jimmygchen** [comment]
Marking this as `backwards-incompat` as I've bumped the `msrv` to `1.69.0`.

**Review by paulhauner — APPROVED** [review]
Very nice! I like the simplification for getting the genesis validators root.

**Comment by jimmygchen** [comment]
> Very nice! I like the simplification for getting the genesis validators root.

It was @michaelsproul's idea! I like it too ☺️

**Pull request closed without merging**